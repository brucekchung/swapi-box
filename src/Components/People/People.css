.People {
  color: white;
  border: 1px solid yellow;
}