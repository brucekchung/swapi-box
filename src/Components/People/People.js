import React from 'react'
import Card from '../Card/Card'
import './People.css'

const People = ({ peopleData }) => {
  const cards = peopleData.map(card => ( <Card data={card}/> ))

  return (
    <div className="People">
      { cards }
    </div>
  )
}

export default People