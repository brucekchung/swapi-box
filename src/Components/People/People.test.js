import React from 'react'
import { shallow } from 'enzyme'
import People from './People'

describe('People', () => {
  const props = {
    peopleData: [{
      "model": "Digger Crawler", 
      "name": "Sand Crawler",
    }],
    favorite: () => {},
  }

  it('should match the snapshot', () => {
    const wrapper = shallow(<People {...props}/>)
    
    expect(wrapper).toMatchSnapshot()
  })
})