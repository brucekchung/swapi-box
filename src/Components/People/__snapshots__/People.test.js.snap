// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    favorite={[Function]}
    peopleData={
        Array [
            Object {
              "model": "Digger Crawler",
              "name": "Sand Crawler",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          data={
                    Object {
                              "model": "Digger Crawler",
                              "name": "Sand Crawler",
                            }
          }
          favorite={[Function]}
          type="peopleCard"
/>,
      ],
      "className": "People",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "people0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
          },
          "favorite": [Function],
          "type": "peopleCard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
                        Object {
                                    "model": "Digger Crawler",
                                    "name": "Sand Crawler",
                                  }
            }
            favorite={[Function]}
            type="peopleCard"
/>,
        ],
        "className": "People",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "people0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "model": "Digger Crawler",
              "name": "Sand Crawler",
            },
            "favorite": [Function],
            "type": "peopleCard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
